### ISSUE TITLE
`spack style` fails because it uses system python instead of spack-provided one

### ISSUE THREAD

**Issue by manuelakuhn** [LLNL/spack#25180]
### Steps to reproduce

I tried to run `spack style` to check if my new packages follow the spack format. This failed because spack used the system python, which is too old, instead of using the one provided by spack itself.

```console
$ rm -rf ~/.spack  # to maje sure to have a clean setup
$ git clone https://github.com/spack/spack.git
$ cd spack
$ . share/spack/setup-env.sh
$ spack style
==> Error: spack style requires Python 3.6 or later.
$ python --version
Python 2.7.13
$ python3 --version
Python 3.5.3
$ spack install py-flake8
$ spack load py-flake8
$ python --version
Python 3.8.11
$ spack style
==> Error: spack style requires Python 3.6 or later.
```
It should not matter if `python` is installed instead of `py-flake8` because `spack style` gives the same result after running `spack load python`.

This might be related to #25093

### Error message

```console
$ spack --debug --stacktrace style
lib/spack/spack/cmd/__init__.py:121 ==> [2021-08-02-13:36:55.842556] Imported style from built-in commands
lib/spack/spack/cmd/__init__.py:121 ==> [2021-08-02-13:36:55.845994] Imported style from built-in commands
lib/spack/spack/cmd/style.py:363 ==> [2021-08-02-13:36:55.856747] Error: spack style requires Python 3.6 or later.
lib/spack/spack/config.py:989 ==> [2021-08-02-13:36:55.867861] Reading config file $spack/etc/spack/defaults/config.yaml
Traceback (most recent call last):
  File "$spack/lib/spack/spack/main.py", line 774, in main
    return _invoke_command(command, parser, args, unknown)
  File "$spack/lib/spack/spack/main.py", line 502, in _invoke_command
    return_val = command(parser, args)
  File "$spack/lib/spack/spack/cmd/style.py", line 363, in style
    tty.die("spack style requires Python 3.6 or later.")
  File "$spack/lib/spack/llnl/util/tty/__init__.py", line 240, in die
    sys.exit(1)
SystemExit: 1
```

### Information on your system

* **Spack:** 0.16.2-3752-be3e6a0e9b
* **Python:** 3.5.3
* **Platform:** linux-debian9-piledriver
* **Concretizer:** original

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
> It should not matter if python is installed instead of py-flake8 because spack style gives the same result after running spack load python.

When you:
```console
$ . share/spack/setup-env.sh
```
Spack sets the `SPACK_PYTHON` environment variable to point to the interpreter used for Spack. From there on  `${SPACK_PYTHON}` will be used as the interpreter for Spack no matter what is the current `python` in your shell. You can of course change the environment variable value to point to a different interpreter.

**Comment by cyrush** [comment]
I hit a related issue - `spack style` failed to bootstrap b/c it tried to install dependencies into the system python.

**Comment by alalazo** [comment]
This should be solved by #34029 Let me know if it is not (possibly in a new issue).
